src/transformers/pipelines/token_classification.py

```diff
@@ -260,45 +378,31 @@ def group_entities(self, entities: List[dict]) -> List[dict]:
         entity_groups = []
         entity_group_disagg = []
 
-        if entities:
-            last_idx = entities[-1]["index"]
-
         for entity in entities:
-
-            is_last_idx = entity["index"] == last_idx
-            is_subword = self.ignore_subwords and entity["is_subword"]
             if not entity_group_disagg:
-                entity_group_disagg += [entity]
-                if is_last_idx:
-                    entity_groups += [self.group_sub_entities(entity_group_disagg)]
+                entity_group_disagg.append(entity)
                 continue
 
-            # If the current entity is similar and adjacent to the previous entity, append it to the disaggregated entity group
+            # If the current entity is similar and adjacent to the previous entity,
+            # append it to the disaggregated entity group
             # The split is meant to account for the "B" and "I" suffixes
             # Shouldn't merge if both entities are B-type
-            if (
-                (
-                    entity["entity"].split("-")[-1] == entity_group_disagg[-1]["entity"].split("-")[-1]
-                    and entity["entity"].split("-")[0] != "B"
-                )
-                and entity["index"] == entity_group_disagg[-1]["index"] + 1
-            ) or is_subword:
+            bi, tag = entity["entity"].split("-")
+            last_bi, last_tag = entity_group_disagg[-1]["entity"].split("-")
+            # Index might not be available if we aggregate words first.
+            index_agree = entity["index"] == entity_group_disagg[-1]["index"] + 1 if "index" in entity else True
```

Comment:
just sayin `True` when `index` is not available feels dangerous 🧐 
non consecutive entities having matching B-I-tags it could happen.

Reply:
we might have to keep track of first subword index and last subword index of the entity during aggregation.
maybe index could be a tuple 🤔

Reply:
You are perfectly correct, and doing the filtering later should actually remove that check anyway (as there can't be any more non-consecutive entities anymore.

Reply:
oh that's right~ turns out it's a - kill two 🐛 with one 🥌- situation.

Reply:
It led to a bit more consequential change, but should be good now.

We are still filtering special_tokens early on, but `ignore_labels` later, so all contributions should be there.